daily_milestone.go

```diff
@@ -156,15 +155,38 @@ func getMilestones(baseURL string, token string, projectID string) ([]simpleMile
 }
 
 // CreateMilestoneData creates new milestones with title and due date
-func createMilestoneData(advance int) []simpleMilestone {
+func createMilestoneData(advance int, milestoneTime string) []simpleMilestone {
 	today := time.Now().Local()
 	list := []simpleMilestone{}
 	for i := 0; i < advance; i++ {
-		date := today.AddDate(0, 0, i).Format("2006-01-02")
-		milestone := simpleMilestone{}
-		milestone.Title = date
-		milestone.DueDate = date
-		list = append(list, milestone)
+		if milestoneTime == "daily" {
+			date := today.AddDate(0, 0, i).Format("2006-01-02")
+			milestone := simpleMilestone{}
+			milestone.Title = date
+			milestone.DueDate = date
+			list = append(list, milestone)
+		} else if milestoneTime == "weekly" {
+
+			today = today.AddDate(0, 0, 7)
+			year, week := today.ISOWeek()
+			milestoneYear := strconv.Itoa(year)
+			milestoneWeek := strconv.Itoa(week)
+			milestones := milestoneYear + "-w" + milestoneWeek
+			milestone := simpleMilestone{}
+			milestone.Title = milestones
+			milestone.DueDate = milestones
```

Comment:
DueDate needs to be an actual day. So it should be the last day of the week.

Reply:
Also this might also be simplified without the "milestones" variable in between

Reply:
As in last day of the week from the day milestone starts, right?

Reply:
Yeah. So if -01 is the first day of the specific week it would be -07.